File: tests/indexing/parsing/lang/fixtures/java/BodySizes.java
public abstract class BodySizes {

    public abstract int abstractMethod(int a);

    public void emptyBody() {}

    @Deprecated
    @SuppressWarnings("unchecked")
    public <T> List<T> annotatedWithLogic(List<T> items, int limit) {
        List<T> result = new ArrayList<>();
        for (T item : items) {
            if (result.size() >= limit) {
                break;
            }
            result.add(item);
        }
        return result;
    }
}

File: tests/indexing/parsing/lang/fixtures/kotlin/BodySizes.kt
interface BodySizes {

    fun abstractMethod(a: Int): Int

    fun emptyBody() {}

    @Deprecated("use takeFirst instead")
    fun annotatedWithLogic(items: List<Int>, limit: Int): List<Int> {
        val result = mutableListOf<Int>()
        for (item in items) {
            if (result.size >= limit) {
                break
            }
            result.add(item)
        }
        return result
    }
}
